### PR TITLE
Fix copying on supplementary plane characters

### PR DESCRIPTION
**Pull request by a4lg**
When I copy text from PDF displayed on pdf.js, only lower 16-bit of code points are copied (upper bits [that require surrogate pair] are ignored). See an example on issue #10529.

This is because PartialEvaluator.readToUnicode method used classic [String.fromCharCode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode) (which does not support out-of-UCS2 code points) instead of ES6's [String.fromCodePoint](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCodePoint) (which does).

Merging this pull request replaces `String.fromCharCode` to `String.fromCodePoint` where necessary (two of four occurrences in `evaluator.js`).

### CODE REVIEW

**Comment by timvandermeij** [comment]
The Windows failures do not look related to this patch.

@Snuffleupagus Could you review this? Also, would a "text" reference test catch this? If so, it should be added.

**Comment by Snuffleupagus** [comment]
> The Windows failures do not look related to this patch.

Firefox was updated on the bot, this is the "usual" sort of fallout from that.

> Also, would a "text" reference test catch this? If so, it should be added.

Yes it will[1], and a test-case is a requirement for accepting the patch.

---
[1] Try adding `#textLayer=visible` to a viewer with/without the patch, with the `pdfBugEnabled` pref set, and compare the output.

**Comment by a4lg** [comment]
OK, I'm working on the test.

**Comment by a4lg** [comment]
Oh, making the test was harder than I expected (I could not figure out what "text" tests do).
I can ... at least regenerate test PDF file only with **free/open source** fonts ([Source Sans Pro](https://github.com/adobe-fonts/source-sans-pro) and [Hanazono Mincho](http://fonts.jp/hanazono/)): [issue10529.pdf](https://github.com/mozilla/pdf.js/files/2848441/issue10529.pdf)

**Comment by a4lg** [comment]
I read the "Contributing" page again and added a testcase. I'm not sure this is okay though.

**Comment by a4lg** [comment]
I ran `gulp makeref` and noticed that some glyphs are missing on Linux (because of insufficient out-of-UCS2 coverage). I think this is still OK since it (barely) shows missing codepoint value in hex and the image after the fix looks very different.

Here is the reference image before/after the fix on Ubuntu 18.10 + additional fonts (as described in "[Bots](https://github.com/mozilla/pdf.js/wiki/Bots)") + Firefox 65.0.

Before the fix:
issue10529-bug-ref-1

After the fix:
issue10529-fixed-ref-1

I haven't tested it on Windows but I think it will not show missing glyphs because of preinstalled PMingLiU-ExtB font.

File: test/test_manifest.json [hunk 1]
```diff
@@ -1451,6 +1451,13 @@
        "type": "eq",
        "nativeImageDecoderSupport": "none"
     },
+    {  "id": "issue10529",
+       "file": "pdfs/issue10529.pdf",
+       "md5": "1a4d404a137c610ff0c747cbea3b8666",
+       "rounds": 1,
+	   "link": false,
```
Comment (Snuffleupagus) [inline test/test_manifest.json]: Please, no tabs anywhere in the code-base.

Also, please follow https://github.com/mozilla/pdf.js/wiki/Squashing-Commits

Comment (a4lg) [reply test/test_manifest.json]: I didn't notice a tab, thanks.

**Comment by a4lg** [comment]
Fixed code style and squashed.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for fixing this!

**Comment by fgilio** [comment]
Hi! I have a little question: Does this execute when copying text in the viewer or when building the text layer?

**Comment by timvandermeij** [comment]
This runs when the text layer is built and therefore it impacts the way the text can be copied.

**Comment by fgilio** [comment]
Awesome, thanks for clarifying!

**Pull request merged**